modAionImpl/src/org/aion/zero/impl/sync/SyncStats.java

```diff
@@ -339,143 +333,67 @@ public void updateTotalBlockRequestsByPeer(String _nodeId, int _totalBlocks) {
-
-                // calculate the average response time
-                average = calculateAverage(requests, responses);
-
-                if (average >= 0) {
-                    // collect a map of average response times by peer
-                    avgResponseTimeByPeers.put(nodeId, average);
-                    overallAvgPeerResponseTime += average;
-                }
-            }
-
-            overallAvgPeerResponseTime =
-                    avgResponseTimeByPeers.isEmpty()
-                            ? 0d
-                            : overallAvgPeerResponseTime / avgResponseTimeByPeers.size();
-
-            return avgResponseTimeByPeers.entrySet().stream()
-                    .sorted(Map.Entry.comparingByValue())
-                    .collect(
-                            Collectors.toMap(
-                                    Map.Entry::getKey,
-                                    Map.Entry::getValue,
-                                    (e1, e2) -> e2,
-                                    LinkedHashMap::new));
-        } finally {
-            responsesLock.unlock();
-        }
+    public ResponseMgr getHeadersResponseMgr() {
+        return headersResponseMgr;
     }
 
-    /**
-     * Computes the average response time for a peer given the lists of gathered requests and
-     * response times.
-     *
-     * @param requestTimes list of times for requests made
-     * @param responseTimes list of times for responses received
-     * @return the average response time for the request-response cycle
-     */
-    private static double calculateAverage(List<Long> requestTimes, List<Long> responseTimes) {
-        int entries = 0;
-        double sum = 0;
-        int size = Math.min(requestTimes.size(), responseTimes.size());
-
-        // only consider requests that had responses
-        for (int i = 0; i < size; i++) {
-            long request = requestTimes.get(i);
-            long response = responseTimes.get(i);
-
-            // ignore data where the requests comes after the response
-            if (response >= request) {
-                sum += response - request;
-                entries++;
-            }
-        }
-
-        if (entries == 0) {
-            // indicates no data
-            return (double) -1;
-        } else {
-            return Math.ceil(sum / entries);
-        }
+    public ResponseMgr getBodiesResponseMgr() {
+        return bodiesResponseMgr;
     }
 
-    /**
-     * Obtains the overall average response time from all active peer nodes
-     *
-     * @return overall average response time
-     */
-    double getOverallAveragePeerResponseTime() {
-        responsesLock.lock();
-        try {
-            return overallAvgPeerResponseTime;
-        } finally {
-            responsesLock.unlock();
+    public Map<String, Map<String, Pair<Double, Integer>>> getResponseStats() {
+        Map<String, Pair<Double, Integer>> statusStats =
```

Comment:
Without a lock or synchronized objects here the data might be slightly inconsistent. The maps can change while this method is being executed.